src/platform/EFR32/wifi/ethernetif.cpp

```diff
@@ -125,6 +125,16 @@ static void low_level_input(struct netif * netif, uint8_t * b, uint16_t len)
     { /* 60 : LWIP frame alignment */
         len = 60;
     }
+
+    /* Drop packets originated from the same interface and is a multicast destination */
+    if ((netif->hwaddr[0] == b[6] && netif->hwaddr[1] == b[7] && netif->hwaddr[2] == b[8] && netif->hwaddr[3] == b[9] &&
```

Comment:
nit: could this be written somehow easier to parse for correctness? I do not fully know what b and hwaddr is, but maybe something like this:

```cpp
constexpr size_t kHwAddrLen = 6;

const void *origin_mac = b + kHwAddrLen; // is this true? I named this out of thin air
const void *dest_mac = b; 

if ((memcmp(netif->hwaddr, origin_mac, kHwAddrLen) == 0) 
    && (memcmp(netif->dest_mac, kHwAddrLen) != 0)) {
  return;
}
```

use some variable names that make it clear what b[0] and b[6] and such are.

Reply:
Incorporated the comment.